nixos/modules/services/audio/alsa.nix

```diff
@@ -78,7 +77,9 @@ in
 
   ###### implementation
 
-  config = mkIf config.sound.enable {
+  config = {
+      sound.enable = mkDefault !(versionAtLeast config.system.stateVersion "18.03")
+    } // mkIf config.sound.enable {
```

Comment:
I'm not sure, but you _might_ need to wrap the `mkIf config.sound.enable { ... }` in `(...)`.